### PR TITLE
PR on nnlda layer 

### PR DESCRIPTION
**Pull request by advikavs**
## Description

PR on nnlda layer

### CODE REVIEW

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:81]: * arguments are a bad pattern imo. We shouldn't use them in the deepchem codebase.

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:95]: Print statements should be removed before merge

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:37]: Should this method be public or private? Do we expect a user to use this?

**Review by rbharath — COMMENTED** [review]
Good start! Made a couple of preliminary comments

File: deepchem/models/dft/nnxc.py
Comment (advikavs) [reply deepchem/models/dft/nnxc.py:11]: Yes, DQC uses dual inheritance as well.

File: deepchem/models/dft/nnxc.py
Comment (advikavs) [reply deepchem/models/dft/nnxc.py:55]: No, but it is used in HybridXC .

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:18]: There should be a docstring here

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:22]: There should be a docstring here

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:12]: All these layers should be added to the `docs/` folder so they render in the layer documentation

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:29]: Can you add a usage example? There should be an explanation of what LDA is as a DFT method, along with a suitable reference. Aim for at least a few paragraphs of discussion

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:43]: What are the valid values? What does each value mean?

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:44]: What are the valid values for the output? What does each value mean?

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:91]: Can you add a usage example? Can you also extend the explanation of what this layer does?

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:107]: What are the possible values of `xcstr`?

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:6]: This looks like a utility function. Can you copy this in directly from dqc?

**Review by rbharath — COMMENTED** [review]
Getting closer! Main issue is we need a lot more documentation to explain what these layers do.

File: deepchem/models/dft/nnxc.py
Comment (advikavs) [reply deepchem/models/dft/nnxc.py:44]: Currently we only support one mode, so I have removed it. Will add it back in the next iterations.

File: deepchem/models/dft/nnxc.py
Comment (advikavs) [reply deepchem/models/dft/nnxc.py:6]: Are we trying to do that in this iteration? If so, I'm not so sure, but I will look into it.

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:166]: This should be formatted properly as a reference section

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [reply deepchem/models/dft/nnxc.py:156]: Can you list explicitly the values that are allowed here? Better to not force users to read a second reference which may be complicated

**Review by rbharath — COMMENTED** [review]
Almost there. Just a couple more comment improvements to address and we should be ready to merge

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:10]: Can you add more detail on when we would want to use this base class and what capabilities it provides? A good paragraph of explanation would be my rule of thumb

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:30]: Can you add some more basic context? That is, the basic set up that this is a class relevant for density functional theory, a basic explanation of what an exchange correlation functional is?

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:89]: What is the shape of `res`?

File: deepchem/models/dft/nnxc.py
Comment (rbharath) [inline deepchem/models/dft/nnxc.py:113]: Can you add a brief explanation of what `libxc` is?

**Review by rbharath — COMMENTED** [review]
Almost ready to go. Have a few more requests for documentation improvement here

**Review by rbharath — COMMENTED** [review]
LGTM, feel free to merge in once CI is clear

**Comment by advikavs** [comment]
> LGTM, feel free to merge in once CI is clear

The CI is fine. But I don't have merge access, it is blocked. Thanks

**Review by rbharath — APPROVED** [review]
LGTM

**Pull request merged**
